src/composables/settingsSyncer.ts

```diff
@@ -0,0 +1,193 @@
+import { type MaybeRef, onMounted, ref, unref } from 'vue'
+
+import {
+  getKeyDataFromCockpitVehicleStorage,
+  NoPathInBlueOsErrorName,
+  setKeyDataOnCockpitVehicleStorage,
+} from '@/libs/blueos'
+import { isEqual } from '@/libs/utils'
+import { useAlertStore } from '@/stores/alert'
+import { useMainVehicleStore } from '@/stores/mainVehicle'
+
+import { useInteractionDialog } from './interactionDialog'
+
+/**
+ * This composable will keep a setting in sync between the browser's local storage and BlueOS.
+ *
+ * When initialized, it will try to get the value from BlueOS. While BlueOS does not connect, it will use the local
+ * stored value and keep trying to communicate with BlueOS to get it's value.
+ *
+ * Once the connection is stablished, if BlueOS doesn't have a value, it will use the local stored one and update
+ * BlueOS with it. On the other hand, if BlueOS has a value, it will ask the user if they want to use the value from
+ *  BlueOS or the local one. Depending on the user's choice, it will update the local value or BlueOS.
+ *
+ * Once everything is in sync, if the local value changes, it will update the value on BlueOS.
+ * In resume, the initial source of truth is decided by the user, and once everything is in sync, the source of truth
+ *  is the local value.
+ * @param { string } key
+ * @param { T } defaultValue
+ * @returns { RemovableRef<T> }
+ */
+export function useBlueOsStorage<T>(key: string, defaultValue: MaybeRef<T>): RemovableRef<T> {
+  const { showDialog, closeDialog } = useInteractionDialog()
+
+  const primitiveDefaultValue = unref(defaultValue)
+  const currentValue = useStorage(key, primitiveDefaultValue)
+  const finishedInitialFetch = ref(false)
+  let initialSyncTimeout: ReturnType<typeof setTimeout> | undefined = undefined
+  let blueOsUpdateTimeout: ReturnType<typeof setTimeout> | undefined = undefined
+
+  const getVehicleAddress = async (): Promise<string> => {
+    const vehicleStore = useMainVehicleStore()
+
+    while (vehicleStore.globalAddress === undefined) {
+      console.debug('Waiting for vehicle global address on BlueOS sync routine.')
+      await new Promise((r) => setTimeout(r, 1000))
+      // Wait until we have a global address
+    }
+
+    return vehicleStore.globalAddress
+  }
+
+  const askIfUserWantsToUseBlueOsValue = async (): Promise<boolean> => {
+    let useBlueOsValue = true
+
+    const preferBlueOs = (): void => {
+      useBlueOsValue = true
+    }
+
+    const preferCockpit = (): void => {
+      useBlueOsValue = false
+    }
+
+    await showDialog({
+      title: 'Conflict with BlueOS',
+      message: `
+        The value for '${key}' that is currently used in Cockpit differs from the one stored in BlueOS. What do you
+        want to do?
+      `,
+      variant: 'warning',
+      actions: [
+        { text: 'Use the value from BlueOS', action: () => preferBlueOs() },
+        { text: "Keep Cockpit's value", action: () => preferCockpit() },
```

Comment:
```suggestion
        { text: 'Use the value from BlueOS', action: preferBlueOs },
        { text: "Keep Cockpit's value", action: preferCockpit },
```

This should work, right ?